src/typings/kafka.d.ts

```diff
@@ -6,7 +6,15 @@ export interface KafkaConfig {
     rejectUnauthorized: boolean;
   };
   sasl?: {
-    mechanism: 'plain' | 'scram-sha-256' | 'scram-sha-512';
+    mechanism: 'plain';
+    username: string;
+    password: string;
+  } | {
+    mechanism: 'scram-sha-256';
+    username: string;
+    password: string;
+  } | {
+    mechanism: 'scram-sha-512';
```

Comment:
❓ Is this equivalent with the previous? Or did sth change here?

Reply:
It is equivalent...We only changed this because we had some issues with type inference in the Kafka Config [here](https://github.com/Workable/orka/blob/e1bc8c05291ae760a6622fc09e8f96df986b5131/src/initializers/kafka/kafka.ts#L88)